semantic-go/src/Language/Go/Tags.hs

```diff
@@ -34,58 +27,47 @@ class ToTags t where
        )
     => t Loc
     -> m ()
-
-instance (ToTagsBy strategy t, strategy ~ ToTagsInstance t) => ToTags t where
-  tags = tags' @strategy
-
-
-class ToTagsBy (strategy :: Strategy) t where
-  tags'
+  default tags
     :: ( Has (Reader Source) sig m
        , Has (Writer Tags.Tags) sig m
+       , Generic1 t
+       , Tags.GTraversable1 ToTags (Rep1 t)
        )
     => t Loc
     -> m ()
+  tags = gtags
 
-
-data Strategy = Generic | Custom
-
-type family ToTagsInstance t :: Strategy where
-  ToTagsInstance (_ :+: _)              = 'Custom
-  ToTagsInstance Go.FunctionDeclaration = 'Custom
-  ToTagsInstance Go.MethodDeclaration   = 'Custom
-  ToTagsInstance Go.CallExpression      = 'Custom
-  ToTagsInstance _                      = 'Generic
-
-instance ToTagsBy 'Custom Go.FunctionDeclaration where
-  tags' t@Go.FunctionDeclaration
```

Comment:
Instead of defining three classes and a type family to select whether we want to use the generic or a custom implementation for each type…